Src/ILGPU/Backends/IL/ILEmitterExtensions.cs

```diff
@@ -67,10 +67,21 @@ public static class ILEmitterExtensions
                 emitter.Emit(LocalOperation.Load, result);
 
                 emitter.Emit(OpCodes.Ldarg_0);
-                emitter.Emit(OpCodes.Ldflda, field);
-                emitter.EmitCall(field.FieldType.GetMethod(
+
+                var fieldHashCode = field.FieldType.GetMethod(
                     GetHashCodeInfo.Name,
-                    BindingFlags.Public | BindingFlags.Instance));
+                    BindingFlags.Public | BindingFlags.Instance);
+                if (field.FieldType.IsValueType &&
+                    fieldHashCode.DeclaringType != field.FieldType)
```

Comment:
MINOR: This condition can probably be simplified since `SpecializedValue` only accepts `unmanaged` types.
```suggestion
                if (fieldHashCode.DeclaringType != field.FieldType)
```

Reply:
@MoFtZ I was looking forward to it supporting other types as requested in https://github.com/m4rs-mt/ILGPU/issues/415 and https://github.com/m4rs-mt/ILGPU/issues/415

Reply:
We can add support later.

`SpecializedValue` applies the `unmanaged` restriction, and `Action<>` and `Func<>` are reference types, so they cannot be used unless we change `SpecializedValue`, introduce a new type like `SpecalizedFuncValue`, or some other API change.

Also, the original condition would apply boxing if "it is a ValueType and does not match the declaring type". However, wouldn't we want to apply boxing to reference types as well? Easier to deal with it later when we have defined an API, and unit tests to ensure it all works.

Reply:
@MoFtZ AFAIK reference type values are "boxes". Which means the version I made does not handle them correctly. They only need `Ldfld` (and not `Ldflda`). If neither of us will be the one to implement specialized lambdas, the person who would will likely stumble upon this error and get puzzled too.

I will implement this code for reference types, but won't add test coverage.